### ISSUE TITLE
Support for watchosx86 was dropped in 0.2.4?

### ISSUE THREAD

**Issue by MikolajKakol** [ionspin/kotlin-multiplatform-bignum#150]
**Describe the bug**
After upgrade from 0.2.3 to 0.2.4 and updating my project to kotlin 1.4.30 I cannot compile on iOS

**Platform**
 - watchosx86

**Additional context**
Error I've got is:

```
      > No matching variant of com.ionspin.kotlin:bignum:0.2.4 was found. The consumer was configured to find a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native', attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86' but:
          - Variant 'ios32ArmApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'ios_arm32' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86'
          - Variant 'ios64ArmApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'ios_arm64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86'
          - Variant 'iosApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'ios_x64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86'
          - Variant 'jsIrApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component:
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'js' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attribute:
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'watchos_x86')
          - Variant 'jsIrRuntimeElements-published' capability com.ionspin.kotlin:bignum:0.2.4:
              - Incompatible because this component declares a usage of 'kotlin-runtime' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'js' and the consumer needed a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attribute:
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'watchos_x86')
          - Variant 'jsLegacyApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component:
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'js' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attribute:
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'watchos_x86')
          - Variant 'jsLegacyRuntimeElements-published' capability com.ionspin.kotlin:bignum:0.2.4:
              - Incompatible because this component declares a usage of 'kotlin-runtime' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'js' and the consumer needed a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attribute:
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'watchos_x86')
          - Variant 'jvmApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares an API of a component:
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attribute:
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'watchos_x86')
          - Variant 'jvmRuntimeElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a runtime of a component:
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'jvm' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attribute:
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'watchos_x86')
          - Variant 'linuxApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_x64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86'
          - Variant 'linuxArm32HfpApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_arm32_hfp' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86'
          - Variant 'linuxArm64ApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'linux_arm64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86'
          - Variant 'macosX64ApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'macos_x64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86'
          - Variant 'metadataApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component:
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'common' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native'
              - Other compatible attribute:
                  - Doesn't say anything about org.jetbrains.kotlin.native.target (required 'watchos_x86')
          - Variant 'mingwX64ApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'mingw_x64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86'
          - Variant 'mingwX86ApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'mingw_x86' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86'
          - Variant 'tvosArm64ApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'tvos_arm64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86'
          - Variant 'tvosX64ApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'tvos_x64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86'
          - Variant 'watchosArm32ApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_arm32' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86'
          - Variant 'watchosArm64ApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_arm64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86'
          - Variant 'watchosX64ApiElements-published' capability com.ionspin.kotlin:bignum:0.2.4 declares a usage of 'kotlin-api' of a component, as well as attribute 'org.jetbrains.kotlin.platform.type' with value 'native':
              - Incompatible because this component declares a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x64' and the consumer needed a component, as well as attribute 'org.jetbrains.kotlin.native.target' with value 'watchos_x86'

```

**Comment by ionspin** [comment]
Hi @MikolajKakol there were some issues on Sonatype/Maven Central upload side yesterday that caused several attempts at publishing to fail. I just managed to republish it completely under version 0.2.7, so please use that version. I'll keep this issue open for a couple of days so if anybody bumps into this problem they can see the solution.

https://status.maven.org/incidents/z70skgbq8vgc

**Comment by ionspin** [comment]
I might have spoken too soon, after looking into this a bit more it seems that something changed with kotlin 1.3.40 and the the `watchos()` build target is now using watchosX64 instead of watchosX86. I need some time to confirm this, but if this is the case I'll add manually watchosX86 build target and publish it in the snapshot version 0.2.8-SNAPSHOT.

**Comment by ionspin** [comment]
So this is the reason why Kotlin changed the `watchos()` https://youtrack.jetbrains.com/issue/KT-43276
And here is the commit that changed that in 1.4.30 https://github.com/JetBrains/kotlin/commit/d6f54a773076ae566de180581f5fa73e569daf59

I will be adding watchos86 target anyways, it will be available in 0.2.8-SNAPSHOT as soon as GitLab CI finishes after pull request is merged, so in an hour or two if there are no problems along the way.
